src/incidents/hooks/useUpdateIncident.tsx

```diff
@@ -3,15 +3,23 @@ import { queryCache, useMutation } from 'react-query'
 import IncidentRepository from '../../shared/db/IncidentRepository'
 import Incident from '../../shared/model/Incident'
 
+function getResolvedOn(incident: Incident) {
```

Comment:
This works, but makes it a little bit harder to type check. You might want to add some return types here so that it is obvious what state is resolved versus not resolved.